src/partition/tests/ProvidedBoundingBoxTest.cpp

```diff
@@ -357,6 +417,130 @@ BOOST_AUTO_TEST_CASE(TestComputeBoundingBox, * testing::OnSize(4))
   tearDownParallelEnvironment();
 }
 
+BOOST_AUTO_TEST_CASE(TestProvidedReceivedPartitionCommunicate, * testing::OnSize(4))
+{
+  //mesh creation
+  int dimensions = 2;
+  bool flipNormals = true;
+  double safetyFactor = 0.1;
+  bool hasToSend=true;
+  mesh::PtrMesh mesh(new mesh::Mesh("mesh", dimensions, flipNormals));  
+  
+  switch (utils::Parallel::getProcessRank()) {
+  case 0: {
+    Eigen::VectorXd position(dimensions);
+    position <<0.5, 0.0;
+    mesh::Vertex& v1 = mesh->createVertex(position);
+    position << 1.5, 0.0;
+    mesh::Vertex& v2 = mesh->createVertex(position);
+    position <<2.0, 1.0;
+    mesh::Vertex& v3 = mesh->createVertex(position);
+    position << 0.5, 1.0;
+    mesh::Vertex& v4 = mesh->createVertex(position);
+    mesh->createEdge(v1, v2);
+    mesh->createEdge(v2, v3);
+    mesh->createEdge(v3, v4);
+    mesh->createEdge(v4, v1);
+
+    mesh->getConnectedRanks().push_back(0);
+    
+    break;
+  }
+  case 1: {
+    Eigen::VectorXd position(dimensions);
+    position <<2.5, 0.0;
+    mesh::Vertex& v1 = mesh->createVertex(position);
+    position << 3.5, 0.0;
+    mesh::Vertex& v2 = mesh->createVertex(position);
+    position <<3.5, 1.0;
+    mesh::Vertex& v3 = mesh->createVertex(position);
+    position << 2.0, 1.0;
+    mesh::Vertex& v4 = mesh->createVertex(position);
+    mesh->createEdge(v1, v2);
+    mesh->createEdge(v2, v3);
+    mesh->createEdge(v3, v4);
+    mesh->createEdge(v4, v1);
+
+    mesh->getConnectedRanks().push_back(1);
+
+    break;
+  }
+  case 2: {    
+
+    mesh->getConnectedRanks().push_back(0);
+    
+    break;
+  }
+  case 3: {    
+
+    mesh->getConnectedRanks().push_back(1);
+
+    break;
+  }
+  }
+
+  mesh->computeState();
+
+  // create communicatror for master com and bb exchange/com/initial com_map
+  com::PtrCommunication participantCom1 = com::PtrCommunication(new com::SocketCommunication());
+  m2n::DistributedComFactory::SharedPointer distrFactory = m2n::DistributedComFactory::SharedPointer(new m2n::GatherScatterComFactory(participantCom1));
+  m2n::PtrM2N m2n = m2n::PtrM2N(new m2n::M2N(participantCom1, distrFactory));
+  setupM2NEnvironment(m2n);
```

Comment:
Why do you need all that?

Reply:
`setupM2NEnvironment(m2n)` does not make use of the `m2n`. Also, the name is misleading as only the master-slave communication on each side is set up, not the the m2n communication. I think you can remove all that.

Reply:
right! removed!